controllers/cluster_controller.go

```diff
@@ -123,13 +123,17 @@ var ErrNextLoop = utils.ErrNextLoop
 // Reconcile is the operator reconcile loop
 func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	contextLogger, ctx := log.SetupLogger(ctx)
-
-	contextLogger.Debug(fmt.Sprintf("reconciling object %#q", req.NamespacedName))
+	var err error
 
 	defer func() {
-		contextLogger.Debug(fmt.Sprintf("object %#q has been reconciled", req.NamespacedName))
+		if err != nil {
```

Comment:
Errors are already reported here: https://github.com/kubernetes-sigs/controller-runtime/blob/9bc967a711e9538cff4099645dd6df64c9ba8a74/pkg/internal/controller/controller.go#L324

We don't need to report them another time.
Here, we need something to help us to know when a reconciliation loop started and when it finished.
Let me try proposing something else.